spec/requests/edit_user_spec.rb

```diff
@@ -1,184 +0,0 @@
-      expect(SmsSenderNumberChangeJob).to receive(:perform_later).with('+1 (202) 555-1213')
-
-      put_via_redirect(phone_confirmation_path, 'code' => user_session[:phone_confirmation_code])
-      expect(@user.reload.mobile).to eq '+1 (555) 555-5555'
-    end
-
-    it 'does not change the current number if incorrect code is entered' do
-      patch_via_redirect(user_two_factor_authentication_path, 'code' => '12345678')
-
-      expect(@user.reload.mobile).to_not eq '+1 (555) 555-5555'
-    end
-  end
-
-  context 'user changes email to nonexistent email and mobile to existing mobile' do
-    before do
-      sign_in_as_a_valid_user(user)
-      patch_via_redirect(
-        '/users',
-        update_user_profile_form: attrs_with_new_email.merge!(mobile: user_with_mobile.mobile)
-      )
-    end
-
-    it 'lets user know they need to confirm both their new mobile and email' do
-      expect(response.body).
-        to include(
-          'Please enter the code sent to <strong>+1 (202) 555-1213</strong>.'
-        )
-      expect(flash[:notice]).to eq t('devise.registrations.email_and_mobile_need_confirmation')
-      expect(last_email.subject).to eq 'Email confirmation instructions'
-      expect(number_of_emails_sent).to eq 1
-    end
-
-    it_behaves_like 'changing_email'
-
-    it 'calls SmsSenderConfirmationJob' do
-      allow(SmsSenderConfirmationJob).to receive(:perform_later)
-
-      patch_via_redirect(
-        '/users',
-        update_user_profile_form: attrs_with_new_email.merge!(mobile: user_with_mobile.mobile)
-      )
-
-      expect(SmsSenderConfirmationJob).to have_received(:perform_later).
-        with(user_session[:phone_confirmation_code], user_with_mobile.mobile)
-    end
-  end
-
-  context 'user changes email to existing email and mobile to nonexistent mobile' do
-    it 'lets user know they need to confirm both their new mobile and email' do
-      sign_in_as_a_valid_user(user)
-      patch_via_redirect(
-        '/users',
-        update_user_profile_form: attrs_for_new_mobile.merge!(email: user_with_mobile.email)
-      )
-
-      expect(response.body).
-        to include('Please enter the code sent to <strong>+1 (555) 555-5555</strong>.')
-      expect(flash[:notice]).to eq t('devise.registrations.email_and_mobile_need_confirmation')
-      expect(user.reload.mobile).to eq '+1 (202) 555-1212'
-      expect(last_email.subject).to eq t('mailer.email_reuse_notice.subject')
-      expect(number_of_emails_sent).to eq 1
-    end
-
-    it 'calls SmsSenderConfirmationJob' do
-      sign_in_as_a_valid_user(user)
-
-      allow(SmsSenderConfirmationJob).to receive(:perform_later)
-
-      patch_via_redirect(
-        '/users',
-        update_user_profile_form: attrs_for_new_mobile.merge!(email: user_with_mobile.email)
-      )
-      expect(SmsSenderConfirmationJob).to have_received(:perform_later).
-        with(user_session[:phone_confirmation_code], '+1 (555) 555-5555')
-    end
-  end
-end
```

Comment:
We still need most of these tests. The only ones that doesn't apply anymore are the last two, where the user changes both email and mobile at the same time.